assets/js/admin/orders.js

```diff
@@ -11,4 +11,51 @@ jQuery( document ).ready( ( $ ) => {
 
 	'use strict';
 
+	const isCommerceOrder = Boolean( wc_facebook_commerce_orders.is_commerce_order );
+
+	let $form               = $( 'form[id="post"]' );
+	let $orderStatusField   = $( '#order_status' );
+
+
+	/**
+	 * Displays the refund modal on form submit.
+	 *
+	 * @param {Event} event
+	 */
+	function displayRefundModal( event ) {
+
+		event.preventDefault();
+
+		$( '#wc-backbone-modal-dialog .modal-close' ).trigger( 'click' );
+
+		new $.WCBackboneModal.View( {
+			target: 'facebook-for-woocommerce-modal',
+			string: {
+				message: wc_facebook_commerce_orders.refund_modal_message,
+				buttons: wc_facebook_commerce_orders.refund_modal_buttons
+			}
+		} );
+
+		$( document.body )
+			.off( 'wc_backbone_modal_response.facebook_for_commerce' )
+			.on( 'wc_backbone_modal_response.facebook_for_commerce', function() {
+				// copy the value of the modal select to the WC field
+				$( '#refund_reason' ).val( $( '#wc_facebook_refund_reason_modal' ).val() );
+				// submit the form
+				$form.data( 'allow-submit', true ).find( ':submit' ).trigger( 'click' );
+			} );
+	}
+
+
+	$form.on( 'submit', function( event ) {
+
+		if ( ! isCommerceOrder || $form.data('allow-submit') ) {
+			return;
+		}
+
+		if ( 'wc-refunded' === $orderStatusField.val() ) {
```

Comment:
Unless a refunded order can be refunded again (which is a question I also added to this review), I guess you need to check the original status - like you did here https://github.com/facebookincubator/facebook-for-woocommerce/pull/1592/commits/9997deb7b221d6b6551c05d44d5266e185c8140d.

Reply:
Done in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1592/commits/5c1a5fdf677675269db0f5a24f93f5dcaec0f224.